src/linker/ref/Mono.Linker.csproj

```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <RootNamespace>Mono</RootNamespace>
+    <Version>0.2.0.0</Version>
+    <LangVersion>latest</LangVersion>
+    <AssemblyName>illink</AssemblyName>
+    <Description>IL Linker</Description>
+    <TargetFramework>netstandard2.0</TargetFramework>
```

Comment:
Not sure about this one. It probably does not hurt just feels odd

Reply:
Not sure which one you mean - the TFM? IsPackable?

Reply:
TFM: netstandard2.0

Reply:
@ericstj might have some guidance - would you be able to PTAL? In particular, do you have any guidance on picking the right TFM here? The PR description has context.

Reply:
A netstandard2.0 ref lets folks build libraries that depend on this once and then they can run on all the frameworks that support netstandard2.0.  If you provide a netstandard2.0 ref you should make sure you cover *all* frameworks that support netstandard2.0, otherwise folks will hit a FileNotFoundException on the platforms where you don't provide an implementation.

Reply:
Thanks @ericstj. In our case the only expected way to consume a plugin that depends on this ref assembly is by loading the plugin into the linker - and we only ship the linker on netcoreapp3.0 and net471. In that case are there problems with using netstandard2.0 for the ref? (Should we be publishing netcoreapp3.0/net471 ref assemblies instead? That would seem weird to me since the plugins should in theory not depend on platform-specific APIs).

Reply:
I think plugins need to publish for a specific TFM (not netstandard) anyways so that you get all their TFM-specific dependencies.  You might be creating a false promise by giving them a netstandard2.0 ref and implying you can create a runnable netstandard layout.

Reply:
I agree with Eric (we don't need net471) so the TFM should match to illink.task netcore tfm

Reply:
Thanks! So my understanding is there are two issues (inferring a bit here - correct me if I'm wrong):
- We can't be sure that plugins will *only* build against netstandard rather than specific TFMs, so we should instead force them into a specific TFM compatible with our implementation.
- There's an expectation that authors can create runnable layouts for any ref assembly, so we should ship an implementation to avoid violating that expectation.

Based on that, I plan to change the TFM to netcoreapp3.0, and to include the full illink.dll implementation assembly in the package (under lib) in addition to the ref. So this becomes more of a "linker library" package than a pure reference assembly.

The downside is that an author trying to create a portable netstandard plugin has to choose a specific TFM - which doesn't really make a difference since we only run on netcoreapp anyway. We might also get people trying to use this package to create their own linker drivers, which may or may not be feasible depending on the surface area we expose.

Reply:
Focusing on netcore only is totally fine, we don't have any use case for .net framework setup in this scenario